lib/tapioca/compilers/dsl/url_helpers.rb

```diff
@@ -0,0 +1,82 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "rails"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      class UrlHelpers < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(Object)).void }
+        def decorate(root, constant)
+          named_routes = Rails.application.routes.named_routes
+          path_helper_methods = named_routes.path_helpers_module.instance_methods(false)
+          url_helper_methods = named_routes.url_helpers_module.instance_methods(false)
+
+          if constant == GeneratedPathHelpersModule
+            root.create_module("GeneratedPathHelpersModule") do |mod|
+              mod.create_include("ActionDispatch::Routing::UrlFor")
+              mod.create_include("ActionDispatch::Routing::PolymorphicRoutes")
+
+              path_helper_methods.each do |method|
+                mod.create_method(
+                  method.to_s,
+                  parameters: [Parlour::RbiGenerator::Parameter.new("*args", type: "T.untyped")],
+                  return_type: "String"
+                )
+              end
+            end
+          elsif constant == GeneratedUrlHelpersModule
+            root.create_module("GeneratedUrlHelpersModule") do |mod|
+              mod.create_include("ActionDispatch::Routing::UrlFor")
+              mod.create_include("ActionDispatch::Routing::PolymorphicRoutes")
+
+              url_helper_methods.each do |method|
+                mod.create_method(
+                  method.to_s,
+                  parameters: [Parlour::RbiGenerator::Parameter.new("*args", type: "T.untyped")],
+                  return_type: "String"
+                )
+              end
+            end
+          else
+            root.path(constant) do |mod|
+              mod.create_include("GeneratedUrlHelpersModule") if
+                constant.ancestors.include?(GeneratedUrlHelpersModule)
+              mod.create_include("GeneratedPathHelpersModule") if
+                constant.ancestors.include?(GeneratedPathHelpersModule)
+              mod.create_extend("GeneratedUrlHelpersModule") if
+                constant.singleton_class.ancestors.include?(GeneratedUrlHelpersModule)
+              mod.create_extend("GeneratedPathHelpersModule") if
+                constant.singleton_class.ancestors.include?(GeneratedPathHelpersModule)
+            end
```

Comment:
```suggestion
            root.path(constant) do |mod|
	            create_mixins_for(mod, constant, GeneratedUrlHelpersModule)
	            create_mixins_for(mod, constant, GeneratedPathHelpersModule)
            end
```
where the method is defined as:
```ruby
def create_mixins_for(mod, constant, helper_module)
  mod.create_include(helper_module.name) if constant.ancestors.include?(helper_module)
  mod.create_extend(helper_module.name) if constant.singleton_class.ancestors.include?(helper_module)  
end